lib/components/isometric_tile_map_component.dart

```diff
@@ -0,0 +1,149 @@
+import 'dart:ui';
+
+import 'package:flame/components/component.dart';
+
+import '../flame.dart';
+import '../position.dart';
+import '../sprite.dart';
+
+/// This represents an isometric tileset to be used in a tilemap.
+///
+/// It's basically a grid of squares, each square has a tile, in order.
+/// The block ids are calculated going row per row, left to right, top to
+/// bottom.
+///
+/// This class will cache the usage of sprites to improve performance.
+class IsometricTileset {
+  /// The image for this tileset.
+  final Image tileset;
+
+  /// The size of each square block within the image.
+  ///
+  /// The image width and height must be multiples of this number.
+  final int size;
+
+  final Map<int, Sprite> _spriteCache = {};
+
+  IsometricTileset(this.tileset, this.size);
+
+  /// Compute the number of columns the image has
+  /// by using the image width and tile width.
+  int get cols => tileset.width ~/ size;
```

Comment:
Maybe not use the short form, and just write columns?
Maybe there should be one for rows too?